### PR TITLE
reference.md: clarify negation.

### PR DESCRIPTION
**Pull request by nodakai**
It would be great if someone could check my English.

### CODE REVIEW

File: src/doc/reference.md [hunk 1]
```diff
@@ -2752,7 +2756,9 @@ Rust defines the following unary operators. They are all written as prefix opera
 before the expression they apply to.
 
 * `-`
-  : Negation. May only be applied to numeric types.
+  : Negation. Signed integer types and floating-point types support negation. It
+    is an error to apply negation to unsigned types; for example, compiler
```
Comment (rphmeier) [inline src/doc/reference.md]: You'd probably want to say "the compiler" here.

Comment (steveklabnik) [inline src/doc/reference.md]: yup!

**Comment by steveklabnik** [comment]
Just that one minor note, otherwise looks good here.

**Comment by nodakai** [comment]
Thanks for the comments, fixed it

**Pull request merged**